node/src/reactor/main_reactor/tests.rs

```diff
@@ -2292,20 +2305,21 @@ async fn run_rewards_network_scenario(
             let total_current_era_weights = current_era_slated_weights
                 .iter()
                 .fold(U512::zero(), move |acc, s| acc + s.1);
+            let weights_block_idx = if switch_blocks.headers[i - 1].is_genesis() {
+                i - 1
+            } else {
+                i - 2
+            };
             let (previous_era_slated_weights, total_previous_era_weights) =
-                if switch_blocks.headers[i - 1].is_genesis() {
-                    (None, None)
-                } else {
-                    match switch_blocks.headers[i - 2].clone_era_end() {
-                        Some(era_report) => {
-                            let next_weights = era_report.next_era_validator_weights().clone();
-                            let total_next_weights = next_weights
-                                .iter()
-                                .fold(U512::zero(), move |acc, s| acc + s.1);
-                            (Some(next_weights), Some(total_next_weights))
-                        }
-                        _ => panic!("unexpectedly absent era report"),
+                match switch_blocks.headers[weights_block_idx].clone_era_end() {
```

Comment:
I think this alters the logic in a way that will cause a panic when it runs (since it will be trying to get an era_report out of a genesis block). Maybe better not to change anything not directly related to keeping track of rewards in `run_rewards_network_scenario` function now, and dedicate time to it later, it's rather complex.

Reply:
However, the tests do pass in the drone logs/ I'll try to take another look at this, but would still prefer to keep unnecessary changes out so we can make these later in the course of work focused only on this.

Reply:
Actually, this change fixes a weird handling of genesis from before. Nodes actually do sign the genesis block, and we can reward those signatures. If you look at the changes I made, there is still some special handling for genesis: we just take the validators for era 1 as validators for era 0 (because they will always be exactly the same), so there won't be any panic, and genesis will be handled properly.

Also, the tests were failing without those changes, because the amounts for the finality signatures from the genesis era weren't being calculated properly.